evennia/contrib/auditing/server.py

```diff
@@ -0,0 +1,266 @@
+"""
+Auditable Server Sessions:
+Extension of the stock ServerSession that yields objects representing 
+all user input and all system output.
+
+Evennia contribution - Johnny 2017
+"""
+
+import os
+import re
+import socket
+
+from django.utils import timezone
+from django.conf import settings as ev_settings
+from evennia.utils import logger, mod_import, get_evennia_version
+from evennia.server.serversession import ServerSession
+
+# Attributes governing auditing of commands and where to send log objects
+AUDIT_CALLBACK = getattr(ev_settings, 'AUDIT_CALLBACK', None)
+AUDIT_IN = getattr(ev_settings, 'AUDIT_IN', False)
+AUDIT_OUT = getattr(ev_settings, 'AUDIT_OUT', False)
+AUDIT_MASK_IGNORE = set(['@ccreate', '@create'] + getattr(ev_settings, 'AUDIT_IGNORE', []))
+AUDIT_MASK_KEEP_BIGRAM = set(['create', 'connect', '@userpassword'] + getattr(ev_settings, 'AUDIT_MASK_KEEP_BIGRAM', []))
+
+if AUDIT_CALLBACK:
+    try:
+        AUDIT_CALLBACK = mod_import(AUDIT_CALLBACK).output
+        logger.log_info("Auditing module online.")
+        logger.log_info("Recording user input = %s." % AUDIT_IN)
+        logger.log_info("Recording server output = %s." % AUDIT_OUT)
+    except Exception as e:
+        logger.log_err("Failed to activate Auditing module. %s" % e)
+
+class AuditedServerSession(ServerSession):
+    """
+    This class represents a player's session and is a template for
```

Comment:
Good documentation, but the bit about portal/server sessions may be better under a `Notes:` section at the end. For those that know what a Session is, it's better to start documenting the bits unique to this implementation (the sections about the auditing etc).